### PR TITLE
Implemented drop trigger

### PR DESCRIPTION
**Pull request by Hydrocharged**
As discussed, we disallow dropping any triggers that are referenced in other triggers.

### CODE REVIEW

File: sql/plan/ddl.go [hunk 1]
```diff
@@ -308,6 +321,20 @@ func (d *DropTable) RowIter(ctx *sql.Context, row sql.Row) (sql.RowIter, error)
 		}
 	}
 
+	if len(d.triggerNames) > 0 {
+		//TODO: if dropping any triggers fail, then we'll be left in a state where triggers exist for a table that was dropped
```
Comment (zachmu) [inline sql/plan/ddl.go:325]: Also need to sort them here to find the order to drop them. At least a TODO, or implement the sort

File: sql/plan/drop_trigger.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+	triggerDb, ok := d.db.(sql.TriggerDatabase)
+	if !ok {
+		if d.IfExists {
+			return sql.RowsToRowIter(), nil
+		} else {
+			return nil, sql.ErrTriggerDoesNotExist.New(d.TriggerName)
+		}
+	}
+	err := triggerDb.DropTrigger(ctx, d.TriggerName)
+	if err != nil && !sql.ErrTriggerDoesNotExist.Is(err) {
```
Comment (zachmu) [inline sql/plan/drop_trigger.go:63]: Shouldn't be an error if the IfExists flag is set. Need a test for this

**Review by zachmu — APPROVED** [review]
LG, just a couple comments (and address the DDL struct comment from the other PR)

File: sql/plan/ddl.go (hunk 1 above)
Comment (zachmu) [reply sql/plan/ddl.go:325]: Actually that's not necessary, I'm dumb :p

Comment (Hydrocharged) [reply sql/plan/ddl.go:325]: Great catch!! This would have bitten us at some point in the future lol

File: sql/plan/drop_trigger.go (hunk 2 above)
Comment (Hydrocharged) [reply sql/plan/drop_trigger.go:63]: Not sure what you mean here. If the `IF EXISTS` flag is set then it is _not_ an error. One of the tests does cover the `IF EXISTS` on a non-existent trigger as well.

File: sql/plan/ddl.go (hunk 1 above)
Comment (Hydrocharged) [reply sql/plan/ddl.go:325]: It's still worthwhile in the event that a deletion fails. That way the existing triggers are still in a parsable state, so it's definitely a good idea...at least until we get statement-level transaction-style support or whatever that would be called lol

Comment (Hydrocharged) [reply sql/plan/ddl.go:325]: But also, I guess it's okay if the triggers are broken due to deleting a table and a trigger randomly failing deletion. You can (and should) still manually go and delete all of the triggers that didn't delete automatically. So yeah I guess it's overall a non-issue.

**Pull request merged**